state/accountsDB_test.go

```diff
@@ -2790,6 +2795,127 @@ func TestAccountsDB_PrintStatsForRootHash(t *testing.T) {
+
+	t.Run("can not sync missing snapshot node should not put activeDbKey", func(t *testing.T) {
+		t.Parallel()
+
+		trieHashes := make([][]byte, 0)
+		valuesMap := make(map[string][]byte)
+		missingNodeError := errors.New("missing trie node")
+		isMissingNodeCalled := false
+		isSyncError := false
+
+		memDbMock := testscommon.NewMemDbMock()
+		memDbMock.PutCalled = func(key, val []byte) error {
+			require.False(t, bytes.Equal(key, []byte(common.ActiveDBKey)), "should not have put active db key")
+
+			valuesMap[string(key)] = val
+			return nil
+		}
+		memDbMock.GetCalled = func(key []byte) ([]byte, error) {
+			if bytes.Equal(key, []byte(common.ActiveDBKey)) {
+				return []byte(common.ActiveDBVal), nil
+			}
+
+			if len(trieHashes) != 0 && bytes.Equal(key, trieHashes[0]) {
+				isMissingNodeCalled = true
+				return nil, missingNodeError
+			}
+
+			val, ok := valuesMap[string(key)]
+			if !ok {
+				return nil, errors.New("key not found")
+			}
+			return val, nil
+		}
+
+		tr, adb := getDefaultTrieAndAccountsDbWithCustomDB(&testscommon.SnapshotPruningStorerMock{MemDbMock: memDbMock})
+		prepareTrie(tr)
+
+		rootHash, _ := tr.RootHash()
+		trieHashes, _ = tr.GetAllHashes()
+
+		syncer := &mock.AccountsDBSyncerStub{
+			SyncAccountsCalled: func(rootHash []byte) error {
+				isSyncError = true
+				return errors.New("sync error")
+			},
+		}
+		_ = adb.SetSyncer(syncer)
+		adb.SnapshotState(rootHash)
+
+		for tr.GetStorageManager().IsPruningBlocked() {
+			time.Sleep(time.Millisecond * 100)
+		}
+
+		assert.True(t, isMissingNodeCalled)
+		assert.True(t, isSyncError)
+	})
+
+	t.Run("nil syncer should not put activeDbKey", func(t *testing.T) {
+		t.Parallel()
+
+		trieHashes := make([][]byte, 0)
+		valuesMap := make(map[string][]byte)
+		missingNodeError := errors.New("missing trie node")
+		isMissingNodeCalled := false
+
+		memDbMock := testscommon.NewMemDbMock()
+		memDbMock.PutCalled = func(key, val []byte) error {
+			require.False(t, bytes.Equal(key, []byte(common.ActiveDBKey)), "should not have put active db key")
+
+			valuesMap[string(key)] = val
+			return nil
+		}
+		memDbMock.GetCalled = func(key []byte) ([]byte, error) {
+			if bytes.Equal(key, []byte(common.ActiveDBKey)) {
+				return []byte(common.ActiveDBVal), nil
+			}
+
+			if len(trieHashes) != 0 && bytes.Equal(key, trieHashes[0]) {
+				isMissingNodeCalled = true
+				return nil, missingNodeError
+			}
+
+			val, ok := valuesMap[string(key)]
+			if !ok {
+				return nil, errors.New("key not found")
+			}
+			return val, nil
+		}
+
+		tr, adb := getDefaultTrieAndAccountsDbWithCustomDB(&testscommon.SnapshotPruningStorerMock{MemDbMock: memDbMock})
+		prepareTrie(tr)
+
+		rootHash, _ := tr.RootHash()
+		trieHashes, _ = tr.GetAllHashes()
+
+		adb.SnapshotState(rootHash)
+
+		for tr.GetStorageManager().IsPruningBlocked() {
+			time.Sleep(time.Millisecond * 100)
+		}
+
+		assert.True(t, isMissingNodeCalled)
+	})
+}
+
+func prepareTrie(tr common.Trie) {
+	key1 := []byte("key1")
```

Comment:
could have been written as:
```GO
func prepareTrie(tr common.Trie, numKeys int) {  
    for i := 0; i < numKeys; i++{
       key := fmt.Sprintf("key%d", i)
       val := fmt.Sprintf("val%d", i)
        _ = tr.Update([]byte(key), []byte(val))
    }

    _ = tr.Commit()
}
```

in this way we can have very large tries (like 10000 or more keys)

Reply:
Refactored.